ROSNET/Field/FieldValue.cs

```diff
@@ -0,0 +1,146 @@
+                case PrimitiveType.Time:
+                case PrimitiveType.Duration:
+                    return 8;
+                default:
+                    return 0;
+            }
+        }
+
+        /// <summary>
+        /// Creates string with value if printValue is true, else string without value (used for messageDefinition)
+        /// </summary>
+        /// <returns>String with value if printValue is true, else returns string without value</returns>
+        public virtual string ToString(bool printValue)
+        {
+            if (printValue)
+            {
+                return this.ToString(); 
+            }
+            return ($"{DataType} {Name}");
+
+        }
+
+        /// <summary>
+        /// Creates string with value
+        /// </summary>
+        /// <returns>String with value</returns>
+        public override string ToString()
+        {
+            if (this.Value.Length == 0)
+            {
+                return ($"{DataType} {Name} noValue");
+            }
+            return ($"{DataType} {Name} {PrettyValue()}");
+        }
+
+        /// <summary>
+        /// Creates string of Value using datatype
+        /// </summary>
+        /// <returns>String of Value</returns>
+        public string PrettyValue()
+        {
+            switch (this.DataType)
+            {
+                case PrimitiveType.Bool:
+                    return ($"{BitConverter.ToBoolean(this.Value)}");
+                case PrimitiveType.Int8:
+                    return ((int)this.Value.First()).ToString();
+                case PrimitiveType.Uint8:
+                    return ((uint)this.Value.First()).ToString();
+                case PrimitiveType.Byte:
+                    return ($"{(sbyte) this.Value.First()}");
+                case PrimitiveType.Char:
+                    return ((char)this.Value.First()).ToString();
+                case PrimitiveType.Int16:
+                    return ($"{BitConverter.ToInt16(this.Value)}");
+                case PrimitiveType.Uint16:
+                    return ($"{BitConverter.ToUInt16(this.Value)}");
+                case PrimitiveType.Int32:
+                    return ($"{BitConverter.ToInt32(this.Value)}");
+                case PrimitiveType.Uint32:
+                    return ($"{BitConverter.ToUInt32(this.Value)}");
+                case PrimitiveType.Int64:
+                    return ($"{BitConverter.ToInt64(this.Value)}");
+                case PrimitiveType.Uint64:
+                    return ($"{BitConverter.ToUInt64(this.Value)}");
+                case PrimitiveType.Float32:
+                    return ($"{BitConverter.ToSingle(this.Value)}");
+                case PrimitiveType.Float64:
+                    return ($"{BitConverter.ToDouble(this.Value)}");
+                case PrimitiveType.String:
+                    return System.Text.Encoding.Default.GetString(this.Value); ;
+                case PrimitiveType.Time:
```

Comment:
Okay this might be a little controversal, but as far as I can tell `Time` and `Duration` are just built in `message`-definitions, and should therefore _not_ have custom parsing here, rather they should be parsed as any other custom message, just that their definition is always available